src/server.rs

```diff
@@ -281,14 +279,81 @@ impl<State: Send + Sync + 'static> Server<State> {
 
     /// Asynchronously serve the app at the given address.
     #[cfg(feature = "h1-server")]
-    pub async fn listen(self, addr: impl ToSocketAddrs) -> std::io::Result<()> {
+    pub async fn listen(self, addr: impl ToSocketAddrs) -> io::Result<()> {
         let listener = async_std::net::TcpListener::bind(addr).await?;
 
         let addr = format!("http://{}", listener.local_addr()?);
         log::info!("Server is listening on: {}", addr);
-        let mut server = http_service_h1::Server::new(addr, listener.incoming(), self);
+        let mut incoming = listener.incoming();
 
-        server.run().await
+        while let Some(stream) = incoming.next().await {
+            let stream = stream?;
+            let addr = addr.clone();
+            let this = self.clone();
+            task::spawn(async move {
+                let res = async_h1::accept(&addr, stream, |req| async {
+                    let res = this.respond(req).await;
+                    let res = res.map_err(|_| io::Error::from(io::ErrorKind::Other))?;
+                    Ok(res)
+                })
+                .await;
+
+                if let Err(err) = res {
+                    log::error!("async-h1 error", { error: err.to_string() });
```

Comment:
would this always be `io::Error::from(io::ErrorKind::Other).to_string()` because of the map_err on 296?

Reply:
On a second read it seems like only _unhandled_ errors from the server/endpoint/middleware stack are transformed into io::ErrorKind::Other? That's certainly less concerning but we probably still want to log the details. Does this need to return io::Result or can it be a broader http_types::Result, since it potentially is representing any variety of error?